IGC/Compiler/Optimizer/OpenCLPasses/AddressSpaceAliasAnalysis/AddressSpaceAliasAnalysis.cpp

```diff
@@ -23,6 +23,8 @@ IN THE SOFTWARE.
 ============================= end_copyright_notice ===========================*/
 
 #include "llvm/Config/llvm-config.h"
+#include "llvmWrapper/IR/DerivedTypes.h"
+#include "llvmWrapper/Analysis/TargetLibraryInfo.h"
```

Comment:
Possibly LLVM headers got shuffled around and not everything is included in LLVM 12 that were in earlier versions. I got a compiler error because of this. Adding these `#include`s fixed the error but caused the "too many arguments to getTLI() macro" popped up which was then fixed by moving `*M.begin()` out into `Function &F` as explained in the below comment thread.

Reply:
I have fixed it by using only `#include <llvm/Analysis/TargetLibraryInfo.h>` and restoring the original code.